scripts/test_env.sh

```diff
@@ -7,62 +7,66 @@ TEST=${TEST:-"./test"}
 HOST=${HOST:-"127.0.0.1"}
 IROHA2_CONFIG_PATH="$TEST/peers/config.json"
 IROHA2_GENESIS_PATH="$TEST/peers/genesis.json"
+DEFAULT_N_PEERS=4
 
-# TODO: don't hard-code these, instead, generate them.
 declare -A public_keys
-public_keys[iroha0]='ed01201c61faf8fe94e253b93114240394f79a607b7fa55f9e5a41ebec74b88055768b'
-public_keys[iroha1]='ed0120cc25624d62896d3a0bfd8940f928dc2abf27cc57cefeb442aa96d9081aae58a1'
-public_keys[iroha2]='ed0120faca9e8aa83225cb4d16d67f27dd4f93fc30ffa11adc1f5c88fd5495ecc91020'
-public_keys[iroha3]='ed01208e351a70b6a603ed285d666b8d689b680865913ba03ce29fb7d13a166c4e7f1f'
 
 declare -A private_keys
-private_keys[iroha0]='282ed9f3cf92811c3818dbc4ae594ed59dc1a2f78e4241e31924e101d6b1fb831c61faf8fe94e253b93114240394f79a607b7fa55f9e5a41ebec74b88055768b'
-private_keys[iroha1]='3bac34cda9e3763fa069c1198312d1ec73b53023b8180c822ac355435edc4a24cc25624d62896d3a0bfd8940f928dc2abf27cc57cefeb442aa96d9081aae58a1'
-private_keys[iroha2]='1261a436d36779223d7d6cf20e8b644510e488e6a50bafd77a7485264d27197dfaca9e8aa83225cb4d16d67f27dd4f93fc30ffa11adc1f5c88fd5495ecc91020'
-private_keys[iroha3]='a70dab95c7482eb9f159111b65947e482108cfe67df877bd8d3b9441a781c7c98e351a70b6a603ed285d666b8d689b680865913ba03ce29fb7d13a166c4e7f1f'
 
 declare -A p2p_ports
-p2p_ports[iroha0]='1337'
-p2p_ports[iroha1]='1338'
-p2p_ports[iroha2]='1339'
-p2p_ports[iroha3]='1340'
+P2P_STARTING_PORT='1337'
 
 declare -A api_ports
-api_ports[iroha0]='8080'
-api_ports[iroha1]='8081'
-api_ports[iroha2]='8082'
-api_ports[iroha3]='8083'
+API_STARTING_PORT='8080'
 
 declare -A telemetry_ports
-telemetry_ports[iroha0]='8180'
-telemetry_ports[iroha1]='8181'
-telemetry_ports[iroha2]='8182'
-telemetry_ports[iroha3]='8183'
+TELEMETRY_STARTING_PORT='8180'
 
-IROHA_GENESIS_ACCOUNT_PUBLIC_KEY='ed01203f4e3e98571b55514edc5ccf7e53ca7509d89b2868e62921180a6f57c2f4e255'
-IROHA_GENESIS_ACCOUNT_PRIVATE_KEY="{ \"digest_function\": \"ed25519\", \"payload\": \"038ae16b219da35aa036335ed0a43c28a2cc737150112c78a7b8034b9d99c9023f4e3e98571b55514edc5ccf7e53ca7509d89b2868e62921180a6f57c2f4e255\" }"
+declare -A hosts
+
+function generate_p2p_port {
+    P2P_PORT=$(($P2P_STARTING_PORT + $1))
+    p2p_ports[$1]=$P2P_PORT
+}
+
+function generate_api_port {
+    API_PORT=$(($API_STARTING_PORT + $1))
+    api_ports[$1]=$API_PORT
+}
+
+function generate_telemetry_port {
+    TELEMETRY_PORT=$(($TELEMETRY_STARTING_PORT + $1))
+    telemetry_ports[$1]=$TELEMETRY_PORT
+}
+
+function generate_peer_key_pair {
+    mapfile -t -n 3 buffer < <($TEST/kagami crypto -c)
+    public_keys[$1]="${buffer[0]}"
+    private_keys[$1]=$(printf '{"digest_function": "%s", "payload": "%s"}' "${buffer[2]}" "${buffer[1]}")
```

Comment:
Seems like we might be doing something wrong in `kagami` if we need this. Should we adjust the syntax so we can just pipe the output? `kagami crypto -c --json`?